spring/boot2-webflux-autoconfigure/src/main/java/com/linecorp/armeria/spring/web/reactive/ArmeriaReactiveWebServerFactory.java

```diff
@@ -207,7 +220,39 @@ public WebServer getWebServer(HttpHandler httpHandler) {
                 firstNonNull(findBean(DataBufferFactoryWrapper.class), DataBufferFactoryWrapper.DEFAULT);
 
         final Server server = configureService(sb, httpHandler, factoryWrapper, getServerHeader()).build();
-        return new ArmeriaWebServer(server, protocol, address, port, beanFactory);
+        armeriaWebServer = new ArmeriaWebServer(server, protocol, address, port, beanFactory);
+        return armeriaWebServer;
+    }
+
+    private boolean needsToReuseWebServer(int port) {
+        final boolean samePort = isManagementPortEqualsToServerPort();
+        Server existingServer = null;
+        try {
+            existingServer = beanFactory.getBean(Server.class);
+        } catch (NoSuchBeanDefinitionException ignore) {
+        }
+
+        if (samePort || existingServer == null) {
+            return false;
+        } else {
+            final Map<InetSocketAddress, ServerPort> activePorts = existingServer.activePorts();
+
+            // reuse the existing WebServer if it has the same port
+            return activePorts.values().stream()
+                              .anyMatch(serverPort -> serverPort.localAddress().getPort() == port);
+        }
+    }
+
+    private boolean isManagementPortEqualsToServerPort() {
+        final Integer managementPort = environment.getProperty("management.server.port", Integer.class);
+        if (managementPort != null && managementPort < 0) {
```

Comment:
How about calling `ensureValidPort(managementPort)` so that an exception is raised if it's under 0?

Reply:
Also, how about returning early if `managementPort` is `null`?